wpilibNewCommands/src/main/java/edu/wpi/first/wpilibj2/command/MecanumFollowerCommand.java

```diff
@@ -0,0 +1,385 @@
+                                DoubleConsumer rearLeftOutputMetersPerSecond,
+                                DoubleConsumer frontRightOutputMetersPerSecond,
+                                DoubleConsumer rearRightOutputMetersPerSecond,
+                                Subsystem... requirements) {
+    m_trajectory = requireNonNullParam(trajectory, "trajectory", "MecanumFollowerCommand");
+    m_pose = requireNonNullParam(pose, "pose", "MecanumFollowerCommand");
+    m_ks = 0;
+    m_kv = 0;
+    m_ka = 0;
+    m_kinematics = requireNonNullParam(kinematics,
+      "kinematics", "MecanumFollowerCommand");
+
+    m_xdController = requireNonNullParam(xdController,
+      "xdController", "MecanumFollowerCommand");
+    m_ydController = requireNonNullParam(ydController,
+      "xdController", "MecanumFollowerCommand");
+    m_thetaController = requireNonNullParam(thetaController,
+      "thetaController", "MecanumFollowerCommand");
+
+    m_maxWheelVelocityMetersPerSecond = requireNonNullParam(maxWheelVelocityMetersPerSecond,
+      "maxWheelVelocityMetersPerSecond", "MecanumFollowerCommand");
+
+    m_frontLeftController = null;
+    m_rearLeftController = null;
+    m_frontRightController = null;
+    m_rearRightController = null;
+
+    m_currentWheelSpeeds = null;
+
+    m_frontLeftOutput = requireNonNullParam(frontLeftOutputMetersPerSecond,
+      "frontLeftOutput", "MecanumFollowerCommand");
+    m_rearLeftOutput = requireNonNullParam(rearLeftOutputMetersPerSecond,
+      "rearLeftOutput", "MecanumFollowerCommand");
+    m_frontRightOutput =  requireNonNullParam(frontRightOutputMetersPerSecond,
+      "frontRightOutput", "MecanumFollowerCommand");
+    m_rearRightOutput = requireNonNullParam(rearRightOutputMetersPerSecond,
+      "rearRightOutput", "MecanumFollowerCommand");
+    addRequirements(requirements);
+  }
+
+  @Override
+  public void initialize() {
+    var initialState = m_trajectory.sample(0);
+
+    // Sample final pose to get robot rotation
+    m_finalPose = m_trajectory.sample(m_trajectory.getTotalTimeSeconds()).poseMeters;
+
+    var initialXVelocity = initialState.velocityMetersPerSecond
+        * initialState.poseMeters.getRotation().getCos();
+    var initialYVelocity = initialState.velocityMetersPerSecond
+        * initialState.poseMeters.getRotation().getSin();
+
+    m_prevSpeeds = m_kinematics.toWheelSpeeds(
+      new ChassisSpeeds(initialXVelocity, initialYVelocity, 0.0));
+
+    m_timer.reset();
+    m_timer.start();
+  }
+
+  @Override
+  public void execute() {
+    double curTime = m_timer.get();
+    double dt = curTime - m_prevTime;
+
+    var desiredState = m_trajectory.sample(curTime);
+    var desiredPose = desiredState.poseMeters;
+
+    var poseError = desiredPose.relativeTo(m_pose.get());
+
+    double targetXVel = m_xdController.calculate(
+        m_pose.get().getTranslation().getX(),
+        desiredPose.getTranslation().getX());
+
+    //System.out.println(targetXVel);
```

Comment:
These debugging statements should be removed.

Reply:
Oops, havent pushed. Thanks.